llvm/include/llvm/Support/ThreadPool.h

```diff
@@ -227,7 +265,7 @@ class ThreadPool {
 class ThreadPoolTaskGroup {
```

Comment:
could it make sense to have this class also as an interface / abstract class, or with virtual methods to allow them to be overridden?

Reply:
This class has very little functionality: everything is delegated to the pool. The main functionality is a "pointer" for the instance and a RAII destructor.

What part of the API would you customize?

Reply:
I see; personally I wouldn't need to customize this class at this time, thus for me it could be fine to leave it as it is. 
But parhaps in general there could be some use cases where one would want to customize the async(), wait() and destructor methods for a group? E.g. keeping track of the num of tasks in the group and refusing to enqueue a task if the group is already too busy.

Reply:
In this PR, I'm not adding any new functionality or changing the contract of the pool with respect to its users. The only change is the ability to substitue the pool implementation itself, so I rather keep it at this.

But also in general I would keep this kind of feature for when the need arise in LLVM to support this kind of things.
It seems a bit overly speculative to me that we'd need this right now, and also things like " refusing to enqueue a task if the group is already too busy" seems to impact the semantics of the API.
(and even the API signature: the API has to be able to "fail").

Reply:
Yeah you're right, I guess we could always add this if/when needed